caikit_nlp/modules/text_generation/peft_prompt_tuning.py

```diff
@@ -170,48 +174,51 @@ def __del__(self):
     def run(
         self,
         text: str,
-        device: Optional[Union[str, int]] = _DETECT_DEVICE,
-        max_new_tokens=20,
-        min_new_tokens=0,
+        max_new_tokens: Optional[int] = 20,
+        min_new_tokens: Optional[int] = 0,
+        truncate_input_tokens: Optional[int] = 0,
+        decoding_method: Optional[str] = "GREEDY",
```

Comment:
Any thoughts on API alignment for the modules between this and the TGIS remote? Specifically, the TGIS API doesn't use `decoding_method`. Having the two modules diverge, even if transformers/TGIS do, feels a bit strange since the args really are doing the same thing 🤔

Reply:
Yep, We would absolutely add support for TGIS modules for all these. That work is going on in parallel by @tharapalanivel .

Reply:
Cool sounds good. Also might be good to think about whether or not we support TGI somehow - it's a bit weird when the proto specs are a bit different